### PR TITLE
Show specific error for manual review error in new checkout

### PR DESCRIPTION
**Pull request by robinandeer**
## What?

Show specific error for manual review error in new checkout

Add `isManualReviewRequired`-prop to `CheckoutErrorModal`.

Refactor `CheckoutPayment` component to derive error state based on checkout error.

## Why?

We should show a specific error and email address for people that require manual review / fail on debt check.

This already works in the Swedish checkout but was not implemented in the new checkout.

**Ticket(s): [GRW-1496]**

[GRW-1496]: https://hedvig.atlassian.net/browse/GRW-1496?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1162** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1162" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/web-onboarding/1162?utm_source=stack-comment).

File: src/client/pages/Checkout/CheckoutPayment/CheckoutPayment.tsx [hunk 1]
```diff
@@ -216,6 +216,18 @@ const useSubmitFormOnSsnChange = (
...
 }
 
+const getCheckoutErrorState = (error?: ApolloError) => {
+  if (!error) return null
+
+  const isManualReviewRequired = checkIsManualReviewRequired(
+    (error?.graphQLErrors || []) as GraphQLError[],
+  )
+
+  if (isManualReviewRequired) return 'MANUAL_REVIEW_REQUIRED'
```
Comment (alebedev) [inline src/client/pages/Checkout/CheckoutPayment/CheckoutPayment.tsx:226]:
```suggestion
  if (checkIsManualReviewRequired(
    (error?.graphQLErrors || []) as GraphQLError[],
  )) {
      return 'MANUAL_REVIEW_REQUIRED'
  }
```

File: src/client/pages/Checkout/CheckoutPayment/index.tsx [hunk 2]
```diff
@@ -19,7 +19,11 @@ export const Checkout = () => {
     return (
       <>
         <LoadingPage loading />
-        <CheckoutErrorModal isVisible={error !== undefined} onRetry={onRetry} />
+        <CheckoutErrorModal
```
Comment (alebedev) [inline src/client/pages/Checkout/CheckoutPayment/index.tsx:22]: Nit: `isVisible` prop can be replaced with conditional rendering
```
!!error && <CheckoutErrorModal ... />
```

**Review by alebedev — APPROVED** [review]
Looks good, some nit comments inside

File: src/client/pages/Checkout/CheckoutPayment/CheckoutPayment.tsx (hunk 1 above)
Comment (alebedev) [reply src/client/pages/Checkout/CheckoutPayment/CheckoutPayment.tsx:226]: Exactly same amount of information in less code.  I think the main point of having readable variables is documenting code

For example, this contains new info
```
const isManualReviewRequired = someUnclearCheck(...)
```

This, on the other hand, doesn't
```
const isManualReviewRequired = checkIsManualReviewRequired(...)
```

It's purely styling issue, so it's up to whatever you as an author like more

File: src/client/pages/Checkout/CheckoutPayment/index.tsx (hunk 2 above)
Comment (robinandeer) [reply src/client/pages/Checkout/CheckoutPayment/index.tsx:22]: I thought conditional rendering for modals was somehow gonna be less performant than letting the modal itself handle that logic.

However, now I can't find any such references.

Still, we should either do conditional rendering OR pass `isVisible` prop, right?

Comment (alebedev) [reply src/client/pages/Checkout/CheckoutPayment/index.tsx:22]: Forgot it's modal, let's keep original version

**Pull request merged**